### PR TITLE
Lpal 482 add secret scanning action

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose

Add github actions for
- GitLeaks
- Trufflehog

Fixes LPAL-482

## Learning

- Trufflehog action: https://github.com/marketplace/actions/trufflehog-actions-scan
- GitLeaks action: https://github.com/marketplace/actions/gitleaks
 
## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: .github/workflows/trufflehog.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  run-trufflehog:
+    name: Run trufflehog report
+    runs-on: ubuntu-latest
+    steps:
+     - uses: actions/checkout@master
```
Comment (townxelliot) [inline .github/workflows/trufflehog.yml:14]: Does this have to user master instead of v1?

File: .github/workflows/trufflehog.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: trufflehog
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  run-trufflehog:
```
Comment (townxelliot) [inline .github/workflows/trufflehog.yml:10]: Maybe just trufflehog, to be consistent with the format used for the gitleaks job?

File: .github/workflows/gitleaks.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: gitleaks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  gitleaks:
```
Comment (townxelliot) [inline .github/workflows/gitleaks.yml:10]: Should this have a name to be consistent with the trufflehog job?

**Review by townxelliot — CHANGES_REQUESTED** [review]

File: .github/workflows/trufflehog.yml (hunk 1 above)
Comment (williamfalconeruk) [reply .github/workflows/trufflehog.yml:14]: good question - will look into it.

File: .github/workflows/trufflehog.yml (hunk 2 above)
Comment (williamfalconeruk) [reply .github/workflows/trufflehog.yml:10]:
```suggestion
  trufflehog:
```

File: .github/workflows/gitleaks.yml (hunk 3 above)
Comment (williamfalconeruk) [reply .github/workflows/gitleaks.yml:10]:
```suggestion
  gitleaks:
    name: Run gitleaks report
```

File: .github/workflows/trufflehog.yml (hunk 1 above)
Comment (williamfalconeruk) [reply .github/workflows/trufflehog.yml:14]:
```suggestion
     - uses: actions/checkout@v1
```

**Review by townxelliot — APPROVED** [review]

**Pull request merged**